src/bootstrap/main.py

```diff
@@ -104,7 +97,7 @@ def repository_type(url):
 #
 
 
-def init(argv):
```

Comment:
Makes it harder to use the bootstrapper as a library though...

Reply:
If we ever need to use it as a library, we could add that then.

Reply:
1. Why do we need to delete it and add it back when we already have it?
2. One legitimate use for having this as parameters instead of assuming sys.argv is to call these functions from test cases to do a bootstrap in a tmpdir.

Reply:
> Why do we need to delete it and add it back when we already have it?

Are we sure we'll add it back?

> One legitimate use for having this as parameters instead of assuming sys.argv is to call these functions from test cases to do a bootstrap in a tmpdir.

Could we wait until we're sure we'll use it?

Maybe there's a cleaner way to do it too, where application code doesn't have to change for tests. `pytest` might have stuff for mocking command-line arguments.

Reply:
Not a Python expert, but I do think not using the global sysargs is cleaner, passing them as a parameter instead.

Reply:
@carlescufi 
Agreed in C at least, where you get the command-line arguments as arguments to `main()`. Storing them globally would be weird and unexpected there.

In Python, the command-line arguments are never implicitly passed to any startup function, and are accessed at `sys.argv` instead (have to `import sys` to get access to them). `sys.argv` immediately says "the command-line arguments" in idiomatic Python. `sys.argv` is kinda in the same class as `os.environ`, which holds the environment.

Think it comes down to whether the code ever needs to deal with anything but the command-line arguments. It would be simpler if it didn't.

Reply:
@ulfalizer fine, I do understand that. So they are a global system variable to start with. In that case I retract my comment since I did not know that.

Reply:
> > One legitimate use for having this as parameters instead of assuming sys.argv is to call these functions from test cases to do a bootstrap in a tmpdir.
> 
> Could we wait until we're sure we'll use it?

Well, I'm kind of heavily hinting that we *should* use it now in exactly this way ;)

Reply:
@mbolivar 
Could you check if there's a way to avoid it while working on the bootstrap tests? In general, I think it's a bad idea to generalize code for testing if you can handle it within the tests itself. People won't know that it was added for the test.